### PR TITLE
Evaluate flags referenced multiple times at the start of prepare

### PR DESCRIPTION
**Pull request by hakonhall**
Avoid exclusiveAllocation flipping in the middle of a cluster prepare, or any other expressions depending on the make-exclusive or shared-host flags.  This also touched various parts of the autoscaler which I hope I got right.

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/HostCapacityMaintainer.java [hunk 1]
```diff
@@ -62,7 +61,7 @@ public class HostCapacityMaintainer extends NodeRepositoryMaintainer {
 
     private final HostProvisioner hostProvisioner;
     private final ListFlag<ClusterCapacity> preprovisionCapacityFlag;
-    private final BooleanFlag makeExclusiveFlag;
+    private final FlagSource flagSource;
```
Comment (freva) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/HostCapacityMaintainer.java:64]: Not used?

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Limits.java [hunk 2]
```diff
@@ -65,12 +66,12 @@ public NodeResources cap(NodeResources resources) {
         return resources;
     }
 
-    public Limits fullySpecified(ClusterSpec clusterSpec, NodeRepository nodeRepository, ApplicationId applicationId) {
+    public Limits fullySpecified(AllocationParams params, NodeRepository nodeRepository) {
```
Comment (freva) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Limits.java:69]: Use `NodeRepository` in `params`?

**Review by freva — DISMISSED** [review]

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/HostCapacityMaintainer.java (hunk 1 above)
Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/HostCapacityMaintainer.java:64]: Removed

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Limits.java (hunk 2 above)
Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Limits.java:69]: Done

**Review by freva — APPROVED** [review]

**Pull request merged**